pkg/bundle/sync.go

```diff
@@ -173,15 +185,11 @@ func (b *bundle) secretBundle(ctx context.Context, ref *trustapi.SourceObjectKey
 	return string(data), nil
 }
 
-type jksEncoder struct {
-	password []byte
-}
-
 // encodeJKS creates a binary JKS file from the given PEM-encoded trust bundle and password.
 // Note that the password is not treated securely; JKS files generally seem to expect a password
 // to exist and so we have the option for one.
-func (e jksEncoder) encode(trustBundle string) ([]byte, error) {
-	cas, err := util.DecodeX509CertificateChainBytes([]byte(trustBundle))
+func (e jksEncoder) encode() ([]byte, error) {
```

Comment:
Why are these signatures being changed?

Reply:
Hello,
Thank you for a review.
_jksEncoder_ and _pkcs12Encoder_ structures bring all necessary information in corresponding properties:
```
type encoderData struct {
	storePasswd string
	bundleData  string
}

type pkcs12Encoder struct {
	encoderData
}

type jksEncoder struct {
	encoderData
}
```
In the future it might be reasonable to rewrite functions like _populateAdditionalFormatData_ to either accept encoder interface and based on the casting result invoke corresponding `encode()` method or use generics, something like that.

Reply:
I also don't think you need to change the signature of this function. That makes a struct instance locked to the bundle data. Why do you need to change it?

Reply:
I guess it makes code more clear and avoid of copying bundle data between function/method.
The bundle data is tightly coupled to its encoder; imagine you would like to add decode(), transform() methods in the future, then it leads that the bundle data should be returned from one encoder method and immediately consumed by another one.
Let's look how is the bundle data was being used before: put to populate function just only to be put to corresponding encoder's method again.

Reply:
That's pretty common in decorators. When will you use function parameters with your view on this?

Reply:
I agree with you, but at this moment encoder structure consists of 2 fields: password one and data one, what if we need more parameters for encoder, more methods. Ephemeral example - push bundle data into some destination, then encode, so that data should be put as argument twice: to some ’push(data)’ as well as to ’encode(data)’ functions. 
My proposal look pretty good, we operate by encoder structure, it might be pushed anyway, with all necessary parameters. Cool!

Reply:
Colleagues, if you insist to use bundle data as argument to `encode` method, I will rewrite, of course.